snapcraft/internal/lxd.py

```diff
@@ -34,19 +36,36 @@
 _PROXY_KEYS = ['http_proxy', 'https_proxy', 'no_proxy', 'ftp_proxy']
 
 
-class Cleanbuilder:
+def _get_tar_filter(tar_filename):
+    def _tar_filter(tarinfo):
+        fn = tarinfo.name
+        if fn.startswith('./parts/') and not fn.startswith('./parts/plugins'):
+            return None
+        elif fn in ('./stage', './prime', tar_filename):
+            return None
+        elif fn.endswith('.snap'):
+            return None
+        return tarinfo
+    return _tar_filter
 
-    def __init__(self, snap_output, tar_filename, project_options,
-                 remote=None):
+
+class Containerbuild:
+
+    def __init__(self, snap_output, source_folder, project_options,
+                 config, container_name, remote=None):
```

Comment:
Can we pass in a smaller massaged data set instead of the full of the config data structure? A smaller thing maybe called `snap_meta` with just the meta data, could be a method of `config`, so callers would pass `config.get_metadata()`